pkg/aks/create.go

```diff
@@ -110,21 +110,17 @@ func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient
 		}
 	}
 
-	var addonProfiles map[string]*containerservice.ManagedClusterAddonProfile
+	addonProfiles := make(map[string]*containerservice.ManagedClusterAddonProfile, 0)
```

Comment:
I was thinking this would cause a problem limiting at zero, but it doesn't.

Reply:
For maps it does not restrict the size. If you know how many keys you will have it saves a ton of time with allocations.

Reply:
Yeah, it's capacity hint - not size. I believe you can omit this too and it would be the same as passing 0.